### PR TITLE
[WIP] Add WebextManifestTransformer

### LINKED ISSUE #1039: Support WebExtensions manifest.json
# 🙋 feature request

### 🤔 Expected Behavior

Parcel takes a `manifest.json` (not to be confused with normal web app `manifest.json`) as the entry point, and handles the assets (HTML, JS, CSS, image files) defined in there. It copies them, as well as referenced assets inside, to `./dist`. Of course, assets will be processed if necessary, i.e., TS to JS, LESS to CSS.

### 😯 Current Behavior

Parcel only copies the `manifest.json` to `./dist`.

### 💁 Possible Solution

Implements a new Asset handler for this.

### 🔦 Context

When developing an extension for browsers, there are many different files in different places, i.e., `background/index.ts`, `popup/index.html`, `devtools/index.html`, `images/icon32.png`, `inject/inject.less`.... However, they are all defined in the main `manifest.json`, which makes it suitable for being the entry point.

Besides, Parcel does not handle multiple entry points nicely, yet.

### LINKED ISSUE #3362: Parcel 2: Web Manifest transformer
Create the `@parcel/transformer-webmanifest` package in `packages/transforms/webmanifest`. This package is a Parcel 2 transform plugin for web manifests. It should be based on the `WebManifestAsset` from Parcel 1.

The `@parcel/transformer-webmanifest` package is responsible for the following things:

* Parsing JSON
* Collecting dependencies
* Rewriting urls
* Generating JSON

We currently only handle the `.webmanifest` extension, but many people want to be able to use `.json` as well (#235, #1039). In order to support this, we need a way for the HTML transformer to specify that the dependency should be treated as a `webmanifest` file regardless of its actual extension.

### PR DESCRIPTION
**Pull request by Drapegnik**
# ↪️ Pull Request

Depends on #3404, Resolves #3362, Resolves #1039

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by Drapegnik** [comment]
Hi! There is a question: how should I handle _static_ assets that should be just copy-pasted to `dist` directory?
- [`_locales/`](https://developer.chrome.com/extensions/i18n)
- [`storage/schema.json`](https://developer.chrome.com/extensions/manifest/storage)
- [`nacl_modules`](https://developer.chrome.com/extensions/manifest/nacl_modules) ??
- [`dictionaries/`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/dictionaries)

File: packages/configs/default/index.json [hunk 1]
```diff
@@ -31,6 +32,8 @@
     "*.html": "@parcel/packager-html",
     "*.css": "@parcel/packager-css",
     "*.js": "@parcel/packager-js",
+    "*.webmanifest": "@parcel/packager-string",
+    "manifest.json": "@parcel/packager-string",
```
Comment (Drapegnik) [inline packages/configs/default/index.json]: the same as in https://github.com/parcel-bundler/parcel/pull/3404#discussion_r314878203 (to discuss)

**Comment by fregante** [comment]
> Hi! There is a question: how should I handle _static_ assets that should be just copy-pasted to `dist` directory?
> 
>     * [`_locales/`](https://developer.chrome.com/extensions/i18n)

It looks like [parcel-plugin-web-extension](https://github.com/kevincharm/parcel-plugin-web-extension) automatically includes those assets:

https://github.com/kevincharm/parcel-plugin-web-extension/blob/f0a81e539458bf946bf74ff6f333c76a11e7df0c/src/index.js#L9

```js
function WebExtensionPlugin(bundler) {
    bundler.addAssetType('json', require.resolve('./ManifestAsset'))

    const options = bundler.options
    const locales = glob.sync(
        path.resolve(options.rootDir, '_locales/**/messages.json')
    )
    bundler.entryFiles.push(...locales)
}
```

**Comment by fregante** [comment]
What is this PR missing to be merged?

**Comment by wachunei** [comment]
Now that #3404 is closed, should this PR be recreated starting from https://github.com/parcel-bundler/parcel/pull/4313 ?

**Comment by fregante** [comment]
I’m not familiar with the implementation details, but the only thing that this manifest has in common with web manifests is the extension. Other than JSON there’s no overlap. How much code can they really share?

**Comment by 101arrowz** [comment]
I've remade this in #5304. Please check there if you'd like the feature added.

**Pull request closed without merging**